src/js/components/Modal.js

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import _ from 'lodash';
+import {TextAreaAndButton} from './GeneralComponents';
+import {ReplyToTweet} from '../services/httpCall';
+import {Tweet} from './Tweets';
+
+const e = React.createElement;
+
+class Modal extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            modalMsgLength: 0,
+            returnMessage: null,
+            retAppend: "",
+            textBox: null,
+        }
+
+        this.updateReturnMessage = this.updateReturnMessage.bind(this);
+        this.clearTextBox = this.clearTextBox.bind(this);
+    }
+
+    componentDidMount() {
+        let tbClass = document.getElementsByClassName("modalTextArea");
+        this.setState({
+            textBox: tbClass && tbClass[0]
+        })
+    }
+
+    updateUI() {
+        if(this.state.textBox && this.state.textBox.value) {
+            this.setState({
+                modalMsgLength: this.state.textBox.value.length
+            });
+        } else {
+            this.setState({
+                modalMsgLength: 0
+            })
+        }
+
+        this.setState({
+            returnMessage: null,
+            retAppend: ""
+        })
+    }
+
+    clearTextBox() {
+        this.state.textBox.value = '';
+        this.setState({
+            modalMsgLength: 0
+        })
+    }
+
+    updateReturnMessage(statusCode) {
+        if(statusCode) {
+            let success;
+            if(statusCode == 200) {
+                success = true;
+            } else {
+                success = false;
+            }
+
+            this.setState({
+                returnMessage: (success ? "Replied Successfully!" : "Failed to Reply!"),
+                retAppend: (success ? " success" : " error")
```

Comment:
what is retAppend?

Reply:
A string to append to the append message component. Mainly there to change colors using css -> "___.success" { color: green } and "____.error" { color: red }. Will rename for better readability